ml_tools/model/nn_strategy.py

```diff
@@ -535,7 +540,12 @@ def __hash__(self) -> int:
                    )
 
     def _build(self, input_tensor: KerasTensor) -> KerasTensor:
-        x = tf.keras.layers.TimeDistributed(tf.keras.layers.Dense(units=self.units, activation=self.activation))(input_tensor)
+        if len(input_tensor.shape) == 3:
+            x = tf.keras.layers.TimeDistributed(
+                    tf.keras.layers.Dense(units=self.units, activation=self.activation)
+                )(input_tensor)
+        else:
+            x = tf.keras.layers.Dense(units=self.units, activation=self.activation)(input_tensor)
```

Comment:
I don't think this is necessary.  If the `StateSeries` used for training / predicting is `len == 1`, then `TimeDistributed` should effectively do nothing, hence I always wrap things with it.  See comment below regarding the temporal NN class.